lib/ecsx/persistence/ecto.ex

```diff
@@ -0,0 +1,68 @@
+defmodule ECSx.Persistence.Ecto do
+  @behaviour ECSx.Persistence.Behaviour
+
+  require Logger
+
+  alias ECSx.Persistence.Ecto.Components
+  alias ECSx.Persistence.Ecto.Component
+
+  @impl ECSx.Persistence.Behaviour
+  def persist_components(components, _opts \\ []) do
+    {:ok, _} =
+      repo().transaction(fn ->
+        repo().query("TRUNCATE ecsx_components")
+
+        Enum.map(components, fn {module, component_tuple_list} ->
+          Enum.map(component_tuple_list, fn component_tuple ->
+            {:ok, _} =
+              Components.create_component(%{
+                module: Atom.to_string(module),
+                data: tuple_to_string(component_tuple)
+              })
+          end)
+        end)
+      end)
+
+    :ok
+  end
```

Comment:
I wonder if there would be a noticeable difference in performance with `Repo.insert_all/3` 
i.e.
```elixir
now = DateTime.utc_now()

to_insert =
  for {module, component_tuple_list} <- components,
      component_tuple <- component_tuple_list do
    %{module: Atom.to_string(module), data: tuple_to_string(component_tuple)}
  end

Repo.insert_all(Component, to_insert, placeholders: %{inserted_at: now, updated_at: now})
```

Reply:
On a note unrelated to `ecsx_persistence_ecto` but inspired by thinking about persistence performance, I think blocking the ECSx Manager during persistence might become problematic.  Maybe in the Manager, we fetch the components, then cast them (not waiting for the results) to a different process which does the file writing, repo inserting, etc.  With our modest default tick rate of 20, that only gives us ~25ms for persistence before it starts delaying the next System tick.

Another thought is capturing persistence metrics and displaying them on the LiveDashboard alongside System times.  I had ruled that out earlier because the persistence doesn't run every tick, but maybe it would be prudent to show the worst-case stats, i.e. instead of telling the users that load above 50% is bad, include the persistence load in the total as if it *were* running every tick.

Reply:
Yeah I can see the insert_all being a potential performance increase in the future. I'd like to see how this performs for now, and we can revisit if we start seeing issues. We do [pull the components](https://github.com/ecsx-framework/ECSx/blob/master/lib/ecsx/persistence.ex#L4-L12) synchronously and then the actual persist happens in a [cast](https://github.com/ecsx-framework/ECSx/blob/master/lib/ecsx/persistence/server.ex#L9-L11).

I am working on a Live Dashboard update to include persistence metrics, hope to have that up in a PR shortly.